client/authority-discovery/src/worker.rs

```diff
@@ -618,6 +643,77 @@ fn hash_authority_id(id: &[u8]) -> libp2p::kad::record::Key {
 	libp2p::kad::record::Key::new(&libp2p::multihash::Sha2_256::digest(id))
 }
 
+// Makes sure all values are the same and returns it
+//
+// Returns Err(_) if not all values are equal. Returns Ok(None) if there are
+// no values.
+fn single<T>(values: impl IntoIterator<Item = T>) -> std::result::Result<Option<T>, ()>
```

Comment:
```suggestion
fn single<T>(values: impl IntoIterator<Item = T>) -> Result<Option<T>, ()>
```

:thinking:

Reply:
Nope. `Result` is hidden by the 1-arg one defined with the crate's own error :shrug: